File: 練習問題/発展問題/データ構造/stack.ts
const stack: string[] = [];

const push = (item: string, pile: string[]): void => {
    pile.push(item);
    console.log('push: ' + item);
};

const pop = (pile: string[]): void => {
    if (pile.length === 0) {
        console.log('pileが空になりました');
        return;
    }
    const item = pile.pop();
    console.log('pop: ' + item);
};

push('A', stack);
push('B', stack);
push('C', stack);
console.log(stack);
pop(stack);
pop(stack);
push('D', stack);
pop(stack);

File: 練習問題/発展問題/データ構造/Stack.java
import java.util.ArrayList;
import java.util.List;

public class Stack {
    static void push(String string, List<String> pile) {
        pile.add(string);
        System.out.println(pile);
    }

    static void pop(List<String> pile) {
        if (pile.isEmpty()) {
            System.err.println("リストが空です");
            return;
        }
        pile.removeLast();
        System.out.println(pile);
    }

    public static void main(String[] args) {
        List<String> pile = new ArrayList<>();
        push("A", pile);
        push("B", pile);
        push("C", pile);
        pop(pile);
        pop(pile);
        push("D", pile);
        pop(pile);
    }
}
